### PR TITLE
13 - refactor img into Image component

### PR DESCRIPTION
**Pull request by jagodakubicka**
- Replaced `` with Next.js `<Image />` component for improved performance and optimization.
- Updated next.config.mjs to include multiple trusted domains for images, as recommended by Next.js for enhanced security.
- Implemented responsive sizing using the `sizes` attribute in the `<Image />` component temporarily. Will need to be adjusted during component styling.

### CODE REVIEW

File: next.config.mjs
Comment (WitchDevelops) [inline next.config.mjs:1]: In the terminal I see this:
⚠ The "images.domains" configuration is deprecated. Please use "images.remotePatterns" configuration instead.

It's a relatively new thing, but let's keep it fresh :)
StackOverflow re this issue: https://stackoverflow.com/questions/77447587/the-images-domains-configuration-is-deprecated-please-use-images-remotepatte

**Review by WitchDevelops — CHANGES_REQUESTED** [review]

File: next.config.mjs
Comment (jagodakubicka) [reply next.config.mjs:1]: Fixed! 😊 Thank you for pointing that out!

File: src/components/BasicInfo.tsx [hunk 1]
```diff
@@ -15,12 +15,16 @@ export const BasicInfo = ({ pet }: any) => {
     >
       <div
         className="rounded-xl shadow-md aspect-auto overflow-hidden"
-        style={{ gridColumn: '1/2' }}
+        style={{ gridColumn: '1/2', maxWidth: '200px' }}
```
Comment (WitchDevelops) [inline src/components/BasicInfo.tsx:18]: I wonder why do we need maxWidth here, it messes up with the view on big screens...

File: src/components/BasicInfo.tsx [hunk 2]
```diff
@@ -15,12 +15,16 @@ export const BasicInfo = ({ pet }: any) => {
...
       >
-        <img className="object-cover" src={pet.img} />
-        {/* TODO: use <Image> */}
-        {/* problem: unconfigured host, look into that */}
-        {/* <Image src={pet.img} alt={pet.name} width='100' height='200' /> */}
+        <div className="relative w-full h-full">
```
Comment (WitchDevelops) [inline src/components/BasicInfo.tsx:20]: What's the reason behind using relative positioning?

Observation:
pet image is not being displayed on small screens

**Review by WitchDevelops — CHANGES_REQUESTED** [review]
There are visual issues with how the pet image is being displayed both on mobile and desktop.

on mobile: image it's not displayed at all
on desktop: image is too small, plenty of white space left, it doesn't look good... let's keep the view where the image on left-hand side fills all the available space in the column.

**Review by WitchDevelops — APPROVED** [review]

**Pull request merged**